docs/4.x/tracking-java.md

```diff
@@ -0,0 +1,169 @@
+
+## Ecommerce Items
+
+Matomo supports ecommerce tracking, i.e. tracking of ecommerce actions like adding items to the cart, removing items from the cart, and purchasing items. Every ecommerce action can be tracked as a goal. To track ecommerce actions, you need to set a goal id in the request. You can set the goal id using the method `MatomoRequest.request().goalId(int)`. The goal id is the id of the goal in Matomo. You can find the goal id in the Matomo user interface under "Administration" -> "Websites" -> "Goals". The goal id is the number in the URL of the edit page.
+
+Ecommerce items can be added to a request using the method `MatomoRequest.request().ecommerceItems(EcommerceItems)`. The items will be added to the request URL as a JSON representation in the query parameter `ec_items`. The JSON representation is an array of JSON objects. Each JSON object represents an ecommerce item. Ecommerce items are only supported if a goal id is set in the request. If you set ecommerce items without a goal id, the items will be ignored by Matomo.
+
+To create an ecommerce items object, you can use the builder `EcommerceItems.builder()`:
+
+```java
+EcommerceItems ecommerceItems = EcommerceItems
+    .builder()
+    .item(EcommerceItem
+        .builder()
+        .sku("XYZ12345")
+        .name("Matomo - The big book about web analytics")
+        .category("Education & Teaching")
+        .price(23.1)
+        .quantity(2)
+        .build())
+    .item(EcommerceItem
+        .builder()
+        .sku("B0C2WV3MRJ")
+        .name("Matomo for data visualization")
+        .category("Education & Teaching")
+        .price(15.0)
+        .quantity(1)
+        .build())
+    .build();
+```
+
+## Custom Variables
+
+Matomo supports custom variables. Custom variables are key-value pairs that can be used to track additional information about the visitor. Custom variables can be set in the request using the method `MatomoRequest.request().visitCustomVariables(CustomVariables)`. The custom variables will be added to the request URL as a JSON representation in the query parameter `cvar`. The JSON representation is an array of JSON objects. Each JSON object represents a custom variable.
+
+To create a custom variables object use the constructor `new CustomVariables()` and add custom variables using the method `CustomVariables.add(String, String)`:
+
+```java
+CustomVariables customVariables = new CustomVariables()
+    .add("Type", "Customer")
+    .add("Status", "Logged In");
+```
+
+The variables get the index in the order of their addition. The first variable has the index 1, the second variable has the index 2, and so on. The index is used to identify the variable in the Matomo user interface. You can find the custom variables in the Matomo user interface under "Visitors" -> "Custom Variables". The index is the number in the URL of the custom variables page.
```

Comment:
Just out of curiosity, how would you handle it if you only want to set the custom variable with index 1 and 3? Set 2 with a null/empty value? Would the empty variable then be sent with the request?

Reply:
Oh, yes that an old mistake of mine. I already changed that in the Matomo Java Tracker. I will update the documentation.

Reply:
Done